libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4017,24 +4052,72 @@ void TypeChecker::endVisit(Literal const& _literal)
 						_literal.location(),
 						"Functions that take 2 arguments can only be used as literal suffixes for rational numbers."
 					);
-				else
+				else if (
+					!!dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(0)) &&
+					!!dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(1))
```

Comment:
```suggestion
					dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(0)) &&
					dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(1))
```

I'd actually prefer to have `!= nullptr` there, but it's not our style so, well...

In either case, the `!!` is superfluous.